scripts/gen-eastwest-gateway.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+#
+# Copyright Istio Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+set -euo pipefail
+SINGLE_CLUSTER=0
+while (( "$#" )); do
+  case "$1" in
+    # Node images can be found at https://github.com/kubernetes-sigs/kind/releases
+    # For example, kindest/node:v1.14.0
+    --single-cluster)
+      SINGLE_CLUSTER=1
+      shift
+    ;;
+    --cluster)
+      CLUSTER=$2
+      shift 2
+    ;;
+    --network)
+      NETWORK=$2
+      shift 2
+    ;;
+    --mesh)
+      MESH=$2
+      shift 2
+    ;;
+    -*)
+      echo "Error: Unsupported flag $1" >&2
+      exit 1
+      ;;
+  esac
+done
+# single-cluster installations may need this gateway to allow VMs to get discovery
+# for non-single cluster, we add additional topology information
+SINGLE_CLUSTER="${SINGLE_CLUSTER:-0}"
+if [[ "${SINGLE_CLUSTER}" -eq 0 ]]; then
+  if [[ -z "${CLUSTER:-}" ]] || [[ -z "${NETWORK:-}" ]] || [[ -z "${MESH:-}" ]]; then
+    echo "Must specify either --single-cluster or --mesh, --cluster, and --network."
+    exit 1
+  fi
+fi
```

Comment:
Do we need to warn if user happens to specify all --single-cluster, --mesh, --cluster, and --network?

Reply:
Hopefully we'll have a better method to configure this soon, so I don't want to put too much time into error handling. (Good catch though.)